abi.html

```diff
@@ -1045,8 +1045,8 @@ <h3><a href="#class-types"> 2.4 Non-POD Class Types </a></h3>
 	to nvalign(D) for base classes or
 	to align(D) for data members.
 	Place D at this offset unless doing so would result in two
-	components (direct or indirect) of the same type having the
-	same offset.
+	components (direct or indirect) of the same type (ignoring
+	cv-qualifiers) having the same offset.
```

Comment:
This should be _top-level_ cv-qualifiers.

Reply:
Hm, are you thinking of cv-qualifiers in template arguments? Empty types can't have non-top-level cv-qualifiers in the usual sense (and if they could, they should be ignored here too, because we don't want objects with similar types to have the same address).

Reply:
Being clear about the qualifiers being ignored only being the top-level qualifiers seems like the right thing to do. If for some strange reason we ever add a new fundamental type that can be empty, the right default rule to apply here would be to ignore only the top-level qualifiers.  A type similarity rule would necessarily be specific to the new fundamental type.

Reply:
We can avoid the question entirely and specify the rule in terms of whether the types are similar. Done.